src/messages.rs

```diff
@@ -5,29 +5,50 @@
 
 use serde::{Deserialize, Serialize};
 
-use crate::{verification_key::VerificationKey, SpendAuth};
+use crate::{frost, verification_key::VerificationKey, SpendAuth};
 
 use std::collections::HashMap;
 
 mod constants;
+mod serialize;
+#[cfg(test)]
+mod tests;
 mod validate;
 
 /// Define our own `Scalar` type instead of using `jubjub::Scalar`.
 ///
 /// The serialization design specifies that `Scalar` uses:
 /// "a 32-byte little-endian canonical representation".
-#[derive(Serialize, Deserialize, Debug)]
+#[derive(Serialize, Deserialize, PartialEq, Debug, Copy, Clone)]
 pub struct Scalar([u8; 32]);
 
+impl From<jubjub::Fr> for Scalar {
+    fn from(value: jubjub::Fr) -> Scalar {
+        Scalar(value.to_bytes())
+    }
+}
+
 /// Define our own `AffinePoint` type instead of using `jubjub::AffinePoint`.
 ///
 /// The serialization design specifies that `AffinePoint` uses:
 /// "a 32-byte little-endian canonical representation".
-#[derive(Serialize, Deserialize, Debug)]
+#[derive(Serialize, Deserialize, PartialEq, Debug, Copy, Clone)]
 pub struct AffinePoint([u8; 32]);
 
+impl From<frost::Commitment> for AffinePoint {
+    fn from(value: frost::Commitment) -> AffinePoint {
+        AffinePoint(jubjub::AffinePoint::from(value.0).to_bytes())
+    }
+}
+
+impl From<jubjub::ExtendedPoint> for AffinePoint {
+    fn from(value: jubjub::ExtendedPoint) -> AffinePoint {
+        AffinePoint(jubjub::AffinePoint::from(value).to_bytes())
+    }
+}
+
```

Comment:
> I am unsure of 2 things here:
> ...
> 2- The `marker` at the end of `group_public` field.

You're serializing `VerificationKey<SpendAuth>` directly, but the design says to convert it to and from bytes:
https://github.com/ZcashFoundation/redjubjub/pull/85/files#diff-eb4c7391aa1f02fd5c5b3a11c199bb7a6efe129fb024fc3f31ff1afbd04ee160R310

Just like `messages::Scalar` and `messages::AffinePoint`, create a `messages::VerificationKeySpendAuth` wrapper. Implement the conversions using `From`, using the functions in the design.